### ISSUE TITLE
Integrating more complex algorithms

### ISSUE THREAD

**Issue by montanalow** [smartcorelib/smartcore#155]
One of the nice things about the python ecosystem is that many of the libraries have all adopted the same fit/predict interfaces as scikit, (e.g. xgboost, lightgbm), which allows 3rd party libraries to interact with them all in a more abstract manner. It would be nice if the smartcore api could similarly be extended in Rust for the xgboost bind gen project, but this would have to take place in one or the other library, since 3rd party crates can't impl traits. 

Is there a plan to either implement bindings for some of the existing crates, or otherwise bindgen c/c++ implementations, or implement them from scratch, or just leave them as 3rd party?

**Comment by Mec-iS** [comment]
yes `smartcore` started as an attempt to follow Python libraries' naming and operational interfaces. So any improvement in this direction is welcome.

no it was decided initially as design principle to keep `smartcore` strictly a pure-Rust library. This implies considering a longer timeframe but a more consistent choice in terms of safety.
